flytekit/core/type_engine.py

```diff
@@ -436,7 +437,11 @@ def dict_to_literal_map(
             # The guessed type takes precedence over the type returned by the python runtime. This is needed
             # to account for the type erasure that happens in the case of built-in collection containers, such as
             # `list` and `dict`.
-            python_type = guessed_python_types.get(k, type(v))
+            python_type = guessed_python_types.get(k)
+            if python_type is None:
+                raise user_exceptions.FlyteValueException(k, "The workflow doesn't have this input key.")
```

Comment:
Let's assume that we have a workflow with input `name: str`. If we give workflow other input keys like `n: str` or `name: int`, we should get an error.
Can we always find all the input keys of the workflow in `guessed_python_types`?